pkg/api/store/cluster/cluster_store.go

```diff
@@ -55,15 +57,16 @@ func (t *transformer) transposeGenericConfigToDynamicField(data map[string]inter
 
 		var driver *v3.KontainerDriver
 		driverName := data["genericEngineConfig"].(map[string]interface{})[clusterprovisioner.DriverNameField].(string)
+		// iterate over kontainer drivers to find the one that maps to the genericEngineConfig DriverName ("kd-**") -> "example"
 		for _, candidate := range drivers {
 			if driverName == candidate.Name {
 				driver = candidate
 				break
 			}
 		}
-
-		if driver == nil {
-			return nil, fmt.Errorf("got unknown driver from kontainer-engine: %v", data[clusterprovisioner.DriverNameField])
+		if driver == nil { // driver is nil if it is never assigned
+			logrus.Warnf("Unable to find the kontainer driver %v that maps to %v", driverName, data[clusterprovisioner.DriverNameField])
+			return data, nil // changed in order to prevent hard 500 errors at UI level when driver is not found
```

Comment:
i would drop this comment completely or at least drop the "changed in order to" part, because someone reading this code later is going to have no context of what you mean by "changed"